### ISSUE TITLE
Remove `InMemoryCatalog` from the test-codebase

### ISSUE THREAD

**Issue by Fokko** [apache/iceberg-python#1110]
### Feature Request / Improvement

`InMemoryCatalog` can be replaced by the SqlCatalog with Sqlite as a backend. Less code is more :)

**Comment by kevinjqliu** [comment]
perhaps its a good idea to alias a new `InMemoryCatalog` implementation using SqlCatalog
such as 
https://github.com/apache/iceberg-python/blob/9857107561d2267813b7ce150b01b4e6ac4b3e34/tests/catalog/test_sql.py#L113-L117

**Comment by hussein-awala** [comment]
@Fokko I agree with @kevinjqliu on the alias proposition, I implemented https://github.com/apache/iceberg-python/pull/1140 which makes `InMemoryCatalog` a subclass of `SqlCatalog`, and overrides some methods to simplify the tests.
